tinygrad/tensor.py

```diff
@@ -541,8 +541,9 @@ def min(self, axis=None, keepdim=False): return -((-self).max(axis=axis, keepdim
 
   def mean(self, axis=None, keepdim=False):
     assert all_int(self.shape), "does not support symbolic shape"
-    out = self.sum(axis=axis, keepdim=keepdim)
-    return out.mul(prod(out.shape)/prod(self.shape)) if 0 not in self.shape else out
+    sum_t = self.sum(axis=axis, keepdim=keepdim)
+    x_shape_prod, sum_shape_prod = prod(self.shape), prod(sum_t.shape)
+    return sum_t if sum_shape_prod == 0 else sum_t.div(x_shape_prod / sum_shape_prod)
```

Comment:
do you really need these variables and an extra line? it's less readable
i think `return out.div(prod(self.shape)/prod(out.shape)) if 0 not in out.shape else out` suffices

Reply:
Yes, I agree. I just noticed how similar the new logic could be to the old code. I will change it.